### PR TITLE
Update `BlockDirective` extensions

### PR DESCRIPTION
**Pull request by anferbui**
Bug/issue #, if applicable: N/A 

## Summary

While looking at https://github.com/swiftlang/swift-docc/pull/982, I found that there's a better place to keep the list of directive names which we consider removed from content:
https://github.com/swiftlang/swift-docc/blob/563b958f424395ddcbd1b657bbceb0cf918cc8e0/Sources/SwiftDocC/Model/DocumentationMarkup.swift#L76-L82

We already had a file with lists of directive names so it'll be clearer to keep all of them in the same file.

I also noticed that a lot of directive names were missing from `BlockDirective.allDirectiveNames`. I've updated the array to include all the missing directives which I could find.

This property is used to throw warnings about directives which are not supported in symbol source documentation.

In the future, I think it would be useful to rework this area of the code so that the arrays don't have to be kept manually up to date as that is quite hard to achieve.

## Dependencies

N/A.

## Testing

Ran unit tests.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- ~~[ ] Added tests~~ Functionality should be unchanged.
- [x] Ran the `./bin/test` script and it succeeded
- ~~[ ] Updated documentation if necessary~~  Functionality should be unchanged.

### CODE REVIEW

**Comment by anferbui** [comment]
@swift-ci please test

File: Sources/SwiftDocC/Model/DocumentationMarkup.swift [hunk 1]
```diff
@@ -154,7 +154,7 @@ struct DocumentationMarkup {
                         // Found deprecation notice in the abstract.
                         deprecation = MarkupContainer(directive.children)
                         return
-                    } else if Self.directivesRemovedFromContent.contains(directive.name) {
+                    } else if BlockDirective.directivesRemovedFromContent.contains(directive.name) {
```
Comment (d-ronnqvist) [inline Sources/SwiftDocC/Model/DocumentationMarkup.swift:157]: Please also remove [this duplicate `directivesRemovedFromContent` definition](https://github.com/swiftlang/swift-docc/pull/983/files#diff-88439ca0724189b95f8ea3d93247496ae24a4bc295d52c34fbc8412c5420492fR76-R82)

Comment (anferbui) [reply Sources/SwiftDocC/Model/DocumentationMarkup.swift]: Nice catch, forgot to do that! ✅

**Comment by anferbui** [comment]
@swift-ci  please test

**Comment by anferbui** [comment]
@swift-ci please test

**Review by d-ronnqvist — APPROVED** [review]

**Comment by anferbui** [comment]
@swift-ci please test

**Pull request merged**